app/Models/Foundation/Summit/Registration/PromoCodes/Traits/PrePaidPromoCodeTrait.php

```diff
@@ -156,6 +156,11 @@ public function getNextAvailableTicketPerType(SummitTicketType $ticket_type): ?S
         return null;
     }
 
+    public function isInfinite(): bool
```

Comment:
since prepaid promocode calculates available qty using a different logic (see https://github.com/OpenStackweb/summit-api/blob/895a59ab9534f0c64924b88a4a1a662a684cab92/app/Models/Foundation/Summit/Registration/PromoCodes/Traits/PrePaidPromoCodeTrait.php#L193), isInfinite() should be according to that

Reply:
then only u need to update the base class method isInfinite() to use getQuantityAvailable() instead of accessing the attribute directly and problem solved
u dont need to update the trait